sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CastSuite.scala

```diff
@@ -651,4 +652,54 @@ class CastSuite extends CastSuiteBase {
         checkEvaluation(cast(cast(interval, StringType), YearMonthIntervalType()), period)
       }
   }
+
+  test("SPARK-35768: Take into account year-month interval fields in cast") {
```

Comment:
Need to test `INTERVAL '...' YEAR` and `INTERVAL '...' MONTH` as well.

Reply:
Done

Reply:
@AngersZhuuuu Have your committed the changes? I don't see inputs like `INTERVAL '123' YEAR`

Reply:
> @AngersZhuuuu Have your committed the changes? I don't see inputs like `INTERVAL '123' YEAR`

Done